dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/AbstractBeanDeploymentArchive.java

```diff
@@ -0,0 +1,78 @@
+package com.ibm.ws.cdi.impl.weld;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Set;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.cdi.internal.interfaces.ArchiveType;
+import com.ibm.ws.cdi.internal.interfaces.CDIArchive;
+import com.ibm.ws.cdi.internal.interfaces.WebSphereBeanDeploymentArchive;
+
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * IBM Corporation - initial API and implementation
+ *******************************************************************************/
+//this class exists to be extended by unit tests
+public abstract class AbstractBeanDeploymentArchive implements WebSphereBeanDeploymentArchive {
```

Comment:
I don't like adding extra abstract classes just for unit testing, it adds extra complexity and makes it harder to find the relevant piece of code when debugging.

We should restructure the algorithm so that it's possible to unit test it without doing this.